xla/stream_executor/kernel.h

```diff
@@ -586,8 +586,13 @@ std::unique_ptr<KernelArgsPackedArrayBase> PackKernelArgs(
 template <typename ArgType>
 inline absl::StatusOr<std::unique_ptr<KernelArgsPackedArrayBase>>
 PackKernelArgs(absl::Span<const ArgType> args, uint32_t shared_mem_bytes) {
+#if GOOGLE_CUDA && CUDA_VERSION >= 12010
+  static constexpr int kKernelArgsLimit = 32764;
+#elif GOOGLE_CUDA
+  static constexpr int kKernelArgsLimit = 4096;
```

Comment:
Where does this new number come from?

Reply:
https://developer.nvidia.com/blog/cuda-12-1-supports-large-kernel-parameters/

Reply:
Why 2^15-4? Also, the documentation says bytes. Each parameter is a pointer, which is 4-8 bytes depending on the pointer size. 

To me it sounds like 1024 is the limit assuming the old CUDA version and 4 byte pointers.

Reply:
Wouldn't that be 4095 then? floor(32764/8).